src/ghci/stderr.rs

```diff
@@ -12,24 +12,47 @@ use tokio::io::Lines;
 use tokio::process::ChildStderr;
 use tokio::sync::mpsc;
 use tokio::sync::oneshot;
-use tokio::sync::Mutex;
 use tracing::instrument;
 
-use super::Ghci;
+use super::Mode;
 
 /// An event sent to a `ghci` session's stderr channel.
 #[derive(Debug)]
 pub enum StderrEvent {
     /// Write to the `error_path` (`ghcid.txt`) file, if any.
     Write(oneshot::Sender<()>),
+
+    /// Set the compilation summary to the given string.
+    SetCompilationSummary {
+        summary: String,
+        sender: oneshot::Sender<()>,
+    },
+
+    /// Set the writer's mode.
+    Mode {
+        mode: Mode,
+        sender: oneshot::Sender<()>,
+    },
 }
 
 pub struct GhciStderr {
-    pub ghci: Weak<Mutex<Ghci>>,
```

Comment:
These pointers weren't being used for anything and add complexity so I got rid of them.